core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeSorterSpillReader.java

```diff
@@ -73,7 +73,9 @@ public UnsafeSorterSpillReader(
     }
 
     final InputStream bs =
-        new NioBufferedFileInputStream(file, (int) bufferSizeBytes);
+        new ReadAheadInputStream(
+            new NioBufferedFileInputStream(file, (int) bufferSizeBytes),
+            (int)bufferSizeBytes, (int)bufferSizeBytes / 2);
```

Comment:
Do we have a test on what would be the best threshold? Should we consider make this configurable?